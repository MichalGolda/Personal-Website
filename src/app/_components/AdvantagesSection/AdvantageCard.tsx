export type AdvantageCardProps = {
  title: string;
  content: string;
  icon: React.ReactElement;
};

const AdvantageCard: React.FC<AdvantageCardProps> = ({
  title,
  content,
  icon,
  ...props
}) => {
  return (
    <div className="flex flex-col gap-y-4 text-left justify-left" {...props}>
      {icon}
      <h4 className="font-bold text-lg md:text-xl lg:text-2xl text-secondary">
        {title}
      </h4>
      <p className="text-sm md:text-base text-body">{content}</p>
    </div>
  );
};

export default AdvantageCard;
