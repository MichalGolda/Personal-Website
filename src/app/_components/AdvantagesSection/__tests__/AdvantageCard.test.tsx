import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react"
import AdvantageCard from "../AdvantageCard"

describe("AdvantageCard component", () => {
    it("renders without errors", () => {
        render(<AdvantageCard iconSrc="/test.png" title="test title" content="test content" data-testid="advantage-card" />)

        const card = screen.getByTestId("advantage-card")
        const icon = screen.getByRole("img")
        const title = screen.getByText("test title")
        const content = screen.getByText("test content")

        expect(card).toBeInTheDocument()

        expect(icon).toBeInTheDocument()
        expect(icon).toHaveAttribute("src", "/_next/image?url=%2Ftest.png&w=128&q=75")
        expect(icon).toHaveAttribute("alt", "test title")
        
        expect(title).toBeInTheDocument()
        expect(title.tagName).toBe("H4")
        
        expect(content).toBeInTheDocument()
        expect(content.tagName).toBe("P")
    })

    it("renders unchanged", () => {
        const { container } = render(<AdvantageCard iconSrc="/test.png" title="test title" content="test content" data-testid="advantage-card" />)
        
        expect(container).toMatchSnapshot()
    })
})
