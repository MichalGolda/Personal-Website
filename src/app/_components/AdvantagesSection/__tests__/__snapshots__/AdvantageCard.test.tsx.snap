// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvantageCard component renders unchanged 1`] = `
<div>
  <div
    class="flex flex-col gap-y-4 text-left justify-left"
  >
    test icon
    <h4
      class="font-bold text-lg md:text-xl lg:text-2xl text-secondary"
    >
      test title
    </h4>
    <p
      class="text-sm md:text-base text-body"
    >
      test content
    </p>
  </div>
</div>
`;
